src/server/job_recov_db.c

```diff
@@ -237,9 +237,11 @@ job_save_db(job *pjob)
 	int rc = -1;
 	int old_mtime, old_flags;
 	char *conn_db_err = NULL;
+	attribute *mtime;
 
-	old_mtime = get_jattr_long(pjob, JOB_ATR_mtime);
-	old_flags = pjob->ji_wattr[JOB_ATR_mtime].at_flags;
+	mtime = get_jattr(pjob, JOB_ATR_mtime);
+	old_mtime = get_attr_l(mtime);
+	old_flags = mtime->at_flags;
```

Comment:
id suggest using object level getters/setters over attribute level setters. You had to do this because we don't have a getter/setter for flags yet. I suggest the following:

old_mtime = get_jattr_long(pjob, JOB_ATR_mtime);
old_flags = (get_jattr(pjob, JOB_ATR_mtime))->at_flags;